run_linux.go

```diff
@@ -2742,14 +2746,27 @@ func (b *Builder) getSSHMount(tokens []string, count int, sshsources map[string]
 }
 
 // cleanupRunMounts cleans up run mounts so they only appear in this run.
-func cleanupRunMounts(mountpoint string, artifacts *runMountArtifacts) error {
+func (b *Builder) cleanupRunMounts(mountpoint string, artifacts *runMountArtifacts) error {
 	for _, agent := range artifacts.Agents {
 		err := agent.Shutdown()
 		if err != nil {
 			return err
 		}
 	}
 
+	//cleanup any mounted images for this run
+	for _, image := range artifacts.MountedImages {
+		if image != "" {
+			i, err := util.LookupImage(b.store, image)
+			// its not necessary we are going to find image, could be stage as well.
```

Comment:
We are re-using `stage` mountpoints they are automatically cleaned up as soon as build is done.

Reply:
Handling and returning error in latest commit. Only ignoring if `image is not found` which means image is already removed or its a stage.

Reply:
If it's not an image, is it a container?  It's a problem that we don't already know what the right thing to do is when we get here.

Reply:
This can produce a different result than it did before we started the command, because an image with a matching name could have been pulled in between.  Keep track of whether or not a name refers to an image, either by not including it in the list of `MountedImages` if it's not actually an image (since we don't need or expect to unmount containers anyway), or by some other means, and do the right thing for it.